hls4ml/report/vivado_report.py

```diff
@@ -309,3 +359,304 @@ def parse_vivado_report(hls_dir):
+
+    time_in_us = time_in_ns / 1000
+    if time_in_us < 1000:
+        return str(time_in_us) + ' \u00B5s'
+
+    time_in_ms = time_in_us / 1000
+    if time_in_ms < 1000:
+        return str(time_in_ms) + ' ms'
+
+    time_in_s = time_in_ms / 1000
+    if time_in_s < 1000:
+        return str(time_in_s) + ' s'
+
+
+def _make_report_body(report_dict, make_table_template, make_header_template):
+    body = ''
+
+    if 'CSynthesisReport' in report_dict:
+        body += make_header_template('C Synthesis report')
+        perf_rows = {
+            'Best-case latency': 'best_latency',
+            'Worst-case latency': 'worst_latency',
+            'Interval Min': 'interval_min',
+            'Interval Max': 'interval_max',
+            'Estimated Clock Period': 'estimated_clock',
+        }
+        area_rows = {
+            'BRAM_18K': 'bram',
+            'DSP48E': 'dsp',
+            'FF': 'ff',
+            'LUT': 'lut',
+            'URAM': 'uram',
+        }
+        body += make_table_template('Performance estimates', perf_rows)
+        body += make_table_template('Resource estimates', area_rows)
+
+        csynth_report = report_dict['CSynthesisReport']
+        target_clock = float(csynth_report['TargetClockPeriod'])
+        best_latency = int(csynth_report['BestLatency'])
+        worst_latency = int(csynth_report['BestLatency'])
+        bram = int(csynth_report['BRAM_18K'])
+        avail_bram = int(csynth_report['AvailableBRAM_18K'])
+        dsp = int(csynth_report['DSP48E'])
+        avail_dsp = int(csynth_report['AvailableDSP48E'])
+        ff = int(csynth_report['FF'])
+        avail_ff = int(csynth_report['AvailableFF'])
+        lut = int(csynth_report['LUT'])
+        avail_lut = int(csynth_report['AvailableLUT'])
+        if 'URAM' in csynth_report:
+            uram = int(csynth_report['URAM'])
+            avail_uram = int(csynth_report['AvailableURAM'])
+
+        params = {}
+
+        params['best_latency'] = str(best_latency) + ' (' + _convert_cycles_to_time(best_latency, target_clock) + ')'
+        params['worst_latency'] = str(worst_latency) + ' (' + _convert_cycles_to_time(worst_latency, target_clock) + ')'
+        params['interval_min'] = csynth_report['IntervalMin']
+        params['interval_max'] = csynth_report['IntervalMax']
+        params['estimated_clock'] = csynth_report['EstimatedClockPeriod']
+
+        params['bram'] = str(bram) + ' / ' + str(avail_bram) + ' (' + str(round(bram / avail_bram * 100, 1)) + '%)'
+        params['dsp'] = str(dsp) + ' / ' + str(avail_dsp) + ' (' + str(round(dsp / avail_dsp * 100, 1)) + '%)'
+        params['ff'] = str(ff) + ' / ' + str(avail_ff) + ' (' + str(round(ff / avail_ff * 100, 1)) + '%)'
+        params['lut'] = str(lut) + ' / ' + str(avail_lut) + ' (' + str(round(lut / avail_lut * 100, 1)) + '%)'
+        if 'URAM' in csynth_report:
+            params['uram'] = str(uram) + ' / ' + str(avail_uram) + ' (' + str(round(uram / avail_uram * 100, 1)) + '%)'
```

Comment:
from my tests, this can still cause a divide by zero error (because both `uram` and `avail_uram` are 0). We could just explicitly check if `avail_uram > 0` before dividing.